lib/Twig/Test/IntegrationTestCase.php

```diff
@@ -17,27 +17,52 @@
  */
 abstract class Twig_Test_IntegrationTestCase extends PHPUnit_Framework_TestCase
 {
-    abstract protected function getExtensions();
+    /**
+     * @return string
+     */
     abstract protected function getFixturesDir();
 
     /**
-     * @dataProvider getTests
+     * @return Twig_ExtensionInterface[]
      */
-    public function testIntegration($file, $message, $condition, $templates, $exception, $outputs)
+    protected function getExtensions()
     {
-        $this->doIntegrationTest($file, $message, $condition, $templates, $exception, $outputs);
+        return array();
+    }
+
+    /**
+     * @return Twig_SimpleFilter[]
+     */
+    protected function getFilters()
+    {
+        return array();
     }
 
     /**
-     * @dataProvider getLegacyTests
-     * @group legacy
+     * @return Twig_SimpleFunction[]
      */
-    public function testLegacyIntegration($file, $message, $condition, $templates, $exception, $outputs)
```

Comment:
Thanks for explaining, it makes sense to me :) I've moved the legacy tests logic back.